alacritty_terminal/src/grid/storage.rs

```diff
@@ -22,10 +9,35 @@ use crate::index::Line;
 /// Maximum number of invisible lines before buffer is resized
 const TRUNCATE_STEP: usize = 100;
 
+/// An optimized ring buffer for fast indexing and rotation
+///
+/// The [`Storage::rotate`] and [`Storage::rotate_up`] functions are fast
+/// modular additions on the internal `zero` field. As compared with
+/// [`slice::rotate_left`] which must rearrange items in memory.
+///
+/// As a consequence, both [`Index`] and [`IndexMut`] are reimplemented for this
+/// type to account for the zeroth element not always being at the start of the
+/// allocation.
+///
+/// Because certain [`Vec`] operations are no longer valid on this type, no
+/// [`Deref`](std::ops::Deref) implementation is provided. Anything from `Vec`
+/// that should be exposed must be done so manually.
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Storage<T> {
     inner: Vec<Row<T>>,
+
+    /// Starting point for the storage of rows
+    ///
+    /// This value represents the starting line offset within the ring buffer. Doing things this
+    /// way allows for scrolling stored lines in a performant way.
+    ///
+    /// The value of this offset may be larger than the `len` itself, and will wrap around to the
+    /// start to form the ring buffer.
     zero: usize,
+
+    /// An **index** separating the visible and scrollback regions
+    ///
+    /// TODO: Why is this needed in storage, and not just in grid.
```

Comment:
I think I've explained this previously, but this is needed for growing/shrinking the available scrollback lines.

Reply:
Yes you have, sorry. I think I simply forgot to remove this `TODO`.